src/SdkCommon/ClientRuntime/ClientRuntime/ExtensibleEnum.cs

```diff
@@ -118,5 +118,17 @@ public override string ToString()
 
             return allKnownValues.ToDictionary(v => v._value, v => v);
         }
+
+        public static T Create(V value)
```

Comment:
This introduces two problems:
  1. Now we need to reason about concurrent modification of the map. This wasn't the case before. That map should be immutable once constructed.
  1. `Activator.CreateInstance` is surely slower than just calling the constructor directly in the generated derived class. What's wrong with that way of doing it?

This doesn't seem worth the effort for what should be a rare case.